### DISCUSSION TITLE
参考文献溢出是正常的吗？

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#136] [Q&A] [answered]
你好，
我发现文后的参考文献的右端，有时候会溢出：
![image](https://user-images.githubusercontent.com/45897471/168405118-2764b118-0f26-4c9b-99a8-ead2594a1b33.png)
像这样情况是正常的吗？
谢谢！

**Comment by zepinglee** [comment]
这里的讨论供开发者参考：<https://github.com/hushidong/biblatex-gb7714-2015/issues/127>

**Reply by earthmoon** [reply]
你好，我看里面的评论，似乎这个方法是最好的：

> `{
> \hyphenpenalty=0 %断词阈值， 值越大越不容易出现断词
> \tolerance=10000 %丑度， 10000为最大无溢出盒子， 参考the texbook 第6章
> \hbadness=10000 %如果丑度超过hbadness这一阀值， 那么就会发出警告
> \printbibliography[heading=bibintoc]
> }`

但是在模板中，我直接把这些代码加到`\begin{document}`之前吗？
谢谢

**Reply by zepinglee** [reply]
建议先把 `biblatex-gb7714-2015` 升级到最新试一下。

`\hyphenpenalty=0` 这些设置应该加到 `\printbibliography` 附近。

**Reply by zepinglee** [reply]
你们的模板似乎不直接使用 `\printbibliography` 命令，这样的设计太不灵活了。

**Reply by earthmoon** [reply]
我在`cls`里面找到了四个`\printbibliography`，感觉这四个似乎都和数字右上角标有关系，而我采用的是作者年格式，所以没有尝试去改。

**Reply by earthmoon** [reply]
我是上两个月安装的latex，适用iso方式安装的。

> 建议先把 `biblatex-gb7714-2015` 升级到最新试一下。
> 
> `\hyphenpenalty=0` 这些设置应该加到 `\printbibliography` 附近。

我这个无法更新，是我的版本已经是最新的还是因为我采用的iso安装的方法造成的？
![image](https://user-images.githubusercontent.com/45897471/168406865-a32e6f81-b6fb-4074-bb6c-8925b9b5bfa0.png)

**Comment by earthmoon** [comment, marked as answer]
这个问题的解决办法（这个操作可以解决这个问题，但是不是严谨的，是不是对的，我不知道，毕竟我看不懂代码）
首先在`cls`文件里找到四个`\printbibliography`，然后在每个`\printbibliography`前面添加两行代码：`\hyphenpenalty=0
\tolerance=10000`，最终的结果如下：
![image](https://user-images.githubusercontent.com/45897471/168951392-531064e4-52ff-4b13-9010-60bd72b0ce24.png)

**Comment by earthmoon** [comment]
这个问题虽然解决了，但是有个问题不明白。`\tolerance=10000`有必要加吗？` \tolerance=10000`现实意义是什么？谢谢。

**Reply by taoxiom** [reply]
看The TeXbook

**Reply by earthmoon** [reply]
> 看The TeXbook

正在看，目前才看到第六章，可以说下具体页码吗？谢谢。

**Reply by sikouhjw** [reply]
> 这个问题虽然解决了，但是有个问题不明白。`\tolerance=10000`有必要加吗？` \tolerance=10000`现实意义是什么？谢谢。

1. 为什么不用 `\sloppy`？
2. 在 https://github.com/hushidong/biblatex-gb7714-2015/issues/127#issuecomment-1099792068 中已经说明该命令在第六章有讲，甚至用 pdf 阅读器的『搜索』功能，一分钟内就能找到答案
   > TEX treats 10000 as if it were “infinite” tolerance,
   > allowing arbitrarily wide space; thus, a tolerance of 10000 will never produce an
   > overfull box, unless something strange occurs like an unhyphenatable word that
   > is wider than the column itself.

**Reply by earthmoon** [reply]
> > 这个问题虽然解决了，但是有个问题不明白。`\tolerance=10000`有必要加吗？` \tolerance=10000`现实意义是什么？谢谢。
> 
> 1. 为什么不用 `\sloppy`？
> 2. 在 [行溢出、行对齐问题-Random "Overfull \hbox" issues hushidong/biblatex-gb7714-2015#127 (comment)](https://github.com/hushidong/biblatex-gb7714-2015/issues/127#issuecomment-1099792068) 中已经说明该命令在第六章有讲，甚至用 pdf 阅读器的『搜索』功能，一分钟内就能找到答案
>    > TEX treats 10000 as if it were “infinite” tolerance,
>    > allowing arbitrarily wide space; thus, a tolerance of 10000 will never produce an
>    > overfull box, unless something strange occurs like an unhyphenatable word that
>    > is wider than the column itself.

是应该用`\sloppy`，当时没有看到[这个网址](https://github.com/hushidong/biblatex-gb7714-2015/issues/127)。虽然我看了` \tolerance=10000`的命令说明，但是没有理解。关于` \tolerance=10000`和`\badness`的对比理解网上说的也不明白。

**Reply by taoxiom** [reply]
> > 看The TeXbook
> 
> 正在看，目前才看到第六章，可以说下具体页码吗？谢谢。

直接搜索`\tolerance`就行了，14章
![image](https://user-images.githubusercontent.com/73472464/236365203-ca341c16-7944-4a7a-9b42-a6ea511b993b.png)
![image](https://user-images.githubusercontent.com/73472464/236365301-372cf71b-cd08-449b-9077-30db508d3b56.png)
也可以看 TeX impatient
![image](https://user-images.githubusercontent.com/73472464/236365543-48ab34f1-8504-4420-a539-3cd192858174.png)

**Reply by earthmoon** [reply]
> > > 看The TeXbook
> > 
> > 
> > 正在看，目前才看到第六章，可以说下具体页码吗？谢谢。
> 
> 直接搜索`\tolerance`就行了，14章 ![image](https://user-images.githubusercontent.com/73472464/236365203-ca341c16-7944-4a7a-9b42-a6ea511b993b.png) ![image](https://user-images.githubusercontent.com/73472464/236365301-372cf71b-cd08-449b-9077-30db508d3b56.png) 也可以看 TeX impatient ![image](https://user-images.githubusercontent.com/73472464/236365543-48ab34f1-8504-4420-a539-3cd192858174.png)

谢谢！我慢慢看过去。
